### PR TITLE
Re-did count down component

### LINKED ISSUE #1245: Setup Countdown
image

### PR DESCRIPTION
**Pull request by atharvanev**

### CODE REVIEW

**Comment by atharvanev** [comment]
image

File: src/components/user/Countdown.jsx [hunk 1]
```diff
@@ -1,5 +1,85 @@
...
+      const seconds = Math.floor((timeLeft % (1000 * 60)) / 1000);
+      timeLeft <= 0
+        ? setCountdown({ days: 0, hours: 0, minutes: 0, seconds: 0 })
+        : setCountdown({ days, hours, minutes, seconds });
+    }, 1000);
+    return () => clearInterval(interval);
+  }, []);
+
+  return (
+    <div className="inline-block bg-green-400 mr-3 mb-3 rounded-lg font-bold">
```
Comment (shahdivyank) [inline src/components/user/Countdown.jsx]: check the implementation of the countdowns in bearhack and designverse, they map through the digit UI elements to minimize the code

**Comment by atharvanev** [comment]
image

File: src/components/user/Countdown.jsx [hunk 2]
```diff
@@ -1,5 +1,70 @@
...
+    const interval = setInterval(() => {
+      const now = new Date().getTime();
+      const timeLeft = endDate - now;
+      const days = Math.floor(timeLeft / (1000 * 60 * 60 * 24));
+      const hours = Math.floor(
+        (timeLeft % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60)
+      );
+      const minutes = Math.floor((timeLeft % (1000 * 60 * 60)) / (1000 * 60));
+      const seconds = Math.floor((timeLeft % (1000 * 60)) / 1000);
+      timeLeft <= 0
```
Comment (shahdivyank) [inline src/components/user/Countdown.jsx:45]: this should be built into the timeLeft initialization

File: src/data/Config.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ const data = {
   start_url: "/",
   year: "2024",
   date: "November 4th 2024",
+  end_date: "April 10th 2024",
```
Comment (shahdivyank) [inline src/data/Config.js]: should be a time associated as well

File: src/data/Config.js [hunk 4]
```diff
@@ -8,6 +8,8 @@ const data = {
   start_url: "/",
   year: "2024",
   date: "November 4th 2024",
+  end_date: "April 9th 2024",
```
Comment (shahdivyank) [inline src/data/Config.js]: endtime and endgate should be combined and called `end`

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**